solr/core/src/java/org/apache/solr/metrics/prometheus/SolrPrometheusCoreExporter.java

```diff
@@ -0,0 +1,104 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.metrics.prometheus;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreCacheMetric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreHandlerMetric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreHighlighterMetric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreIndexMetric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreMetric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreNoOpMetric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreSearcherMetric;
+import org.apache.solr.metrics.prometheus.core.SolrCoreTlogMetric;
+
+/**
+ * This class maintains a {@link io.prometheus.metrics.model.snapshots.MetricSnapshot}s exported
+ * from solr.core {@link com.codahale.metrics.MetricRegistry}
+ */
+public class SolrPrometheusCoreExporter extends SolrPrometheusExporter {
+  public final String coreName;
+  public final boolean cloudMode;
+  public static final String ADMIN = "ADMIN";
+  public static final String QUERY = "QUERY";
+  public static final String UPDATE = "UPDATE";
+  public static final String REPLICATION = "REPLICATION";
+  public static final String TLOG = "TLOG";
+  public static final String CACHE = "CACHE";
+  public static final String SEARCHER = "SEARCHER";
+  public static final String HIGHLIGHTER = "HIGHLIGHTER";
+  public static final String INDEX = "INDEX";
+  public static final String CORE = "CORE";
+
+  public SolrPrometheusCoreExporter(String coreName, boolean cloudMode) {
+    super();
+    this.coreName = coreName;
+    this.cloudMode = cloudMode;
+  }
+
+  /**
+   * Export {@link Metric} to {@link io.prometheus.metrics.model.snapshots.MetricSnapshot} and
+   * registers the Snapshot
+   *
+   * @param dropwizardMetric the {@link Meter} to be exported
+   * @param metricName Dropwizard metric name
+   */
+  @Override
+  public void exportDropwizardMetric(Metric dropwizardMetric, String metricName) {
+    SolrCoreMetric solrCoreMetric = categorizeCoreMetric(dropwizardMetric, metricName);
+    solrCoreMetric.parseLabels().toPrometheus(this);
+  }
+
+  private SolrCoreMetric categorizeCoreMetric(Metric dropwizardMetric, String metricName) {
+    String metricCategory = metricName.split("\\.")[0];
```

Comment:
I'd rather we don't split merely to get the first component. 
An alternative approach here is to have many startsWith checks so that we're not even parsing/extracting.  I leave it to you to pick a path; either are fine I think.

Reply:
Personally more a fan of split instead of many `startsWith`. I'll keep as if unless someone feels strongly otherwise.

Reply:
split takes a "limit" second arg.

Reply:
Set limit to 2 and only grab the `[0]` index.